pkg/workspace/operator.go

```diff
@@ -28,6 +28,60 @@ var (
 	ErrWorkspaceAlreadyExist = errors.New("workspace has already existed")
 )
 
+// CheckWorkspaceExistenceByDefaultOperator checks the workspace exists or not by default operator.
+func CheckWorkspaceExistenceByDefaultOperator(name string) (bool, error) {
+	operator, err := NewValidDefaultOperator()
+	if err != nil {
+		return false, err
+	}
+	return operator.WorkspaceExist(name), nil
+}
+
+// GetWorkspaceByDefaultOperator gets a workspace by default operator.
+func GetWorkspaceByDefaultOperator(name string) (*workspace.Workspace, error) {
+	operator, err := NewValidDefaultOperator()
+	if err != nil {
+		return nil, err
+	}
+	return operator.GetWorkspace(name)
+}
+
+// GetWorkspaceNamesByDefaultOperator list all the workspace names by default operator.
+func GetWorkspaceNamesByDefaultOperator() ([]string, error) {
+	operator, err := NewValidDefaultOperator()
+	if err != nil {
+		return nil, err
+	}
+	return operator.GetWorkspaceNames()
+}
+
+// CreateWorkspaceByDefaultOperator creates a workspace by default operator.
+func CreateWorkspaceByDefaultOperator(ws *workspace.Workspace) error {
+	operator, err := NewValidDefaultOperator()
+	if err != nil {
+		return err
+	}
+	return operator.CreateWorkspace(ws)
+}
+
+// UpdateWorkspaceByDefaultOperator updates a workspace by default operator.
+func UpdateWorkspaceByDefaultOperator(ws *workspace.Workspace) error {
+	operator, err := NewValidDefaultOperator()
+	if err != nil {
+		return err
+	}
+	return operator.UpdateWorkspace(ws)
+}
+
+// DeleteWorkspaceByDefaultOperator deletes a workspace by default operator
+func DeleteWorkspaceByDefaultOperator(name string) error {
+	operator, err := NewValidDefaultOperator()
+	if err != nil {
+		return err
+	}
+	return operator.DeleteWorkspace(name)
+}
+
 // Operator is used to handle the CURD operations of workspace. Operator only supports local file
 // system as backend for now.
 type Operator struct {
```

Comment:
Why not define the Operator as an interface as we will have various types of operators in the future?

Reply:
That's a good suggestion! But it's for the future, now it's more visualized.
Now it seems C/U/R/D defines the functions of the interface "Operator", but if there is any more functions need, we can get more feedback from may-exist users and subsequent R&D work.
Besides, even though the operator becomes an interface, there still needs a struct holding the structure of the configuration of the Operator, which may have two field `backend type` and `backend config`, and it's a struct.
Defining Operator an interface is a good idea, but I think we can do it when there are more workspace operator backends, now the struct is ok. What's your opinion?